src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -261,6 +267,12 @@ public JobHostTraceConfiguration Tracing
         }
 
         /// <summary>
+        /// get host-level metadata which the extension may Read to do configuration. 
+        /// </summary>
+        [Obsolete("preview")]
+        public JObject HostConfigMetadata { get; set; }
```

Comment:
>public [](start = 8, length = 6)

This is important.  It will get passed to the extension's Initialize function to let it do what ScriptBinding.Initialize did.